augmentedreality/Common/source/Android/ArCoreWrapper.cpp

```diff
@@ -285,20 +284,41 @@ void ArCoreWrapper::createArSession()
   if (m_arSession)
     return;
 
-  // request camera permission
-  auto checkPermissionResultFuture = QtAndroidPrivate::checkPermission(QtAndroidPrivate::PermissionType::Camera);
-  checkPermissionResultFuture.waitForFinished();
-  const auto checkPermissionResult = checkPermissionResultFuture.result();
-  if (checkPermissionResult != QtAndroidPrivate::PermissionResult::Authorized)
+  QCameraPermission cameraPermission;
+  QEventLoop loop;
+  QTimer timeout;
```

Comment:
The timer is there to prevent a permanent hang in the loop.exec() due to some sort of hiccup or something. The tradeoff is that if the user returns and selects a permission after the timer has ended, the app will crash after. IMO, we should not use a timer, but I wanted to see others thoughts on this. If for some reason the loop.exec() bugs out on a forever loop, the user will just close and re-open the app.